collectors/proc.plugin/proc_pagetypeinfo.c

```diff
@@ -0,0 +1,344 @@
+            if (simple_pattern_matches(filter_types, setid))
+                continue;
+
+            // "Node" + NUMA-NodeID + ZoneName + TypeName
+            char setname[4+1+MAX_ZONETYPE_NAME+1+MAX_PAGETYPE_NAME +1];
+            snprintfz(setname, MAX_ZONETYPE_NAME + MAX_PAGETYPE_NAME, "Node %d %s %s",
+                pgl->node, pgl->zone, pgl->type);
+
+            st_nodezonetype[p] = rrdset_create_localhost(
+                    "mem"
+                    , setid
+                    , NULL
+                    , "pagetype"
+                    , NULL
+                    , setname
+                    , "B"
+                    , PLUGIN_PROC_NAME
+                    , PLUGIN_PROC_MODULE_PAGETYPEINFO_NAME
+                    , NETDATA_CHART_PRIO_MEM_PAGEFRAG + p
+                    , update_every
+                    , RRDSET_TYPE_STACKED
+            );
+            for (o = 0; o < MAX_PAGETYPE_ORDER; o++) {
+                char dimid[3+1];
+                snprintfz(dimid, 3, "%lu", o);
+                char dimname[20+1];
+                dim_name(dimname, o);
+
+                pgl->rd[o] = rrddim_add(st_nodezonetype[p], dimid, dimname, PAGE_SIZE, 1, RRD_ALGORITHM_ABSOLUTE);
+            }
+        }
+    }
+
+    if(unlikely(!cnt_nodes)) {
+        error("PLUGIN: PROC_PAGETPEINFO: Cannot find the number of CPUs in %s", PLUGIN_PROC_MODULE_PAGETYPEINFO_NAME);
+        return 1;
+    }
+
+    // --------------------------------------------------------------------
+    // Update pagelines
+
+    // Process each line
+    p = 0;
+    for (l=4; l<lines; l++) {
+
+        if (!pagetypeinfo_line_valid(ff, l))
+            continue;
+
+        int words = procfile_linewords(ff, l);
+
+        if (words < 6+cnt_pageorders) {
+            error("Unable to read line %lu, only %d words found instead of %d", l, words, 6 + cnt_pageorders);
+            break;
+        }
+
+        for (o = 0; o < MAX_PAGETYPE_ORDER; o++) {
+            // Reset counter
+            if (p == 0)
+                systemorders[o].size = 0;
+
+            // Update orders of the current line
+            pagelines[p].free_pages_size[o] = str2uint64_t(procfile_lineword(ff, l, o+6)) * 1 << o;
+
+            // Update sum by order
+            systemorders[o].size += pagelines[p].free_pages_size[o];
+        }
+
+        p++;
+    }
+
+    // --------------------------------------------------------------------
+    // update RRD values
+
+    // Global system per order
+    if (do_global) {
+        rrdset_next(st_order);
+        for (o = 0; o < MAX_PAGETYPE_ORDER; o++) {
+            rrddim_set_by_pointer(st_order, systemorders[o].rd, systemorders[o].size);
+        }
+        rrdset_done(st_order);
+    }
+
+    // Per Node-Zone-Type
+    if (do_detail) {
+        for (p = 0; p < pagelines_cnt; p++) {
+            // Skip empty graphs
+            if (!st_nodezonetype[p]) {
+                error("Skipping line %lu", p);
```

Comment:
A lot of redundant messages in `error.log`.

Reply:
indeed, not needed here.